aten/src/ATen/native/Fill.cpp

```diff
@@ -106,7 +106,23 @@ Tensor& fill_diagonal_(Tensor& self, Scalar fill_value, bool wrap) {
   return self;
 }
 
+Tensor& zero_cpu_(Tensor &self) {
+  void* ptr = self.data_ptr();
+  if (nullptr == ptr) {
+    return self.fill_(0);
+  }
+  int64_t nelements = at::prod_intlist(self.sizes());
+  int64_t size_bytes = nelements * self.dtype().itemsize();
+  if (size_bytes > 0) {
+    std::memset(ptr, 0, size_bytes);
```

Comment:
We should avoid using `std::memset` to zero out buffers since it can be optimized away by the compiler :)
From https://en.cppreference.com/w/cpp/string/byte/memset:
> std::memset may be optimized away (under the as-if rules) if the object modified by this function is not accessed again for the rest of its lifetime (e.g. gcc bug 8537). For that reason, this function cannot be used to scrub memory (e.g. to fill an array that stored a password with zeroes). Solutions for that include **std::fill with volatile pointers**, **C11 memset_s**, **FreeBSD explicit_bzero** or **Microsoft SecureZeroMemory**.

Reply:
Is gcc actually able to conclude that `ptr` becomes dead at the end of this function? I don't think so: it's owned by the tensor which escapes from this function.